src/main/java/io/github/bobocodebreskul/config/PropertiesConfiguration.java

```diff
@@ -0,0 +1,105 @@
+package io.github.bobocodebreskul.config;
+
+import static java.util.Objects.isNull;
+
+import io.github.bobocodebreskul.context.exception.ConfigurationFileNotFoundException;
+import io.github.bobocodebreskul.context.exception.InvalidPropertyValueException;
+import io.github.bobocodebreskul.context.exception.LoadingPropertiesFailedException;
+import java.io.InputStream;
+import java.util.Properties;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * PropertiesConfiguration class stores properties from configuration file and provides an API to
+ * get property as string or integer value.
+ */
+@Slf4j
+public class PropertiesConfiguration {
+
+  private static String CONFIG_FILE = "application.properties";
+
+  private static Properties properties;
+
+  /**
+   * Method loads properties from configuration file and stores then in properties variable.
+   */
+  public static void loadProperties() {
+    Properties properties = new Properties();
+    try (InputStream input = PropertiesConfiguration.class.getClassLoader()
+        .getResourceAsStream(CONFIG_FILE)) {
+      if (input == null) {
+        log.info("Sorry, unable to find %s".formatted(CONFIG_FILE));
+        throw new ConfigurationFileNotFoundException(
+            "Sorry, unable to find %s".formatted(CONFIG_FILE));
+      }
+
+      properties.load(input);
+      log.info("Properties from %s file was loaded.".formatted(CONFIG_FILE));
+    } catch (Exception e) {
+      log.error("Loading properties from %s file failed.".formatted(CONFIG_FILE), e);
+      throw new LoadingPropertiesFailedException(
+          "Loading properties from %s file failed.".formatted(CONFIG_FILE), e);
+    }
+    PropertiesConfiguration.properties = properties;
+  }
+
+  /**
+   * Method returns property string value loaded from configuration file.
+   *
+   * @param propertyName property key name
+   * @return string property value
+   */
+  public static String getProperty(String propertyName) {
+    log.debug("Get string property \"%s\"".formatted(propertyName));
+    return properties.getProperty(propertyName);
+  }
+
+  /**
+   * Method returns property string value loaded from configuration file or returns default value if
+   * property doesn't exist.
+   *
+   * @param propertyName property key name
+   * @param defaultValue property default value
+   * @return string property value
+   */
+  public static String getPropertyOrDefault(String propertyName, String defaultValue) {
+    log.debug("Get string property \"%s\"".formatted(propertyName));
+    String property = properties.getProperty(propertyName);
+    return isNull(property) ? defaultValue : property;
```

Comment:
You can use Optional API here. 
```
return Optional.ofNullable(properties.getProperty(propertyName)) .orElse(defaultValue);
```

Reply:
I think it does not required wrap with optional only for orElse method it is spend additional resources, and do not give any benefits, but maybe I will use != operator

Reply:
I agree. Creation of the `Optional` object is redundant resources, it's minor but still redundant. 
`Optional` is intended to be used as a return result, of course, it can be used to streamline some code based on `null` but in this exact case I agree with Mykola - the ternary operator looks simpler.